js/cards.js

```diff
@@ -85,15 +93,45 @@ export async function defineCardComponent() {
       return `${CLOUDINARY_URL}/q_auto/${this.data["Rarity Folder"]}/${this.data["Filename"]}.png`;
     }
 
+    setupCardGacha() {
+      // Reset the initial z-index back to max pull
+      card_z_index = HIGHEST_Z_INDEX;
+      // Show the back of the card - unsophisticated style
+      // This is automatically hidden for the collections page, so we have to reset it for the gacha
+      this.back.classList.remove("hidden");
+      // For the gacha page only
+      // Prevent the front from being tabbed on, so that you don't need to tab twice per card
+      this.front.tabIndex = "-1";
+      // Add the event listener to every card
+      setupHover();
+    }
+
     flipCard() {
-      this.front.classList.toggle("hidden");
-      this.back.classList.toggle("hidden");
+      // update the next value for the zIndex
+      card_z_index++;
+
+      // register this as a clicked card, and add the class to animate to flip over
+      this.classList.add("clicked");
+      this.holder.classList.add("flip");
+
+      // Start the animation and update the z-index when the animation starts
+      // This is to stack any opened cards in the reverse order
+      this.addEventListener("animationstart", updateZIndex);
+      this.image.classList.add("opened");
+      this.image.classList.add("animated");
```

Comment:
I wonder if it's better to add the delay in the CSS or by delaying the adding of the class with JavaScript?

Reply:
I've done it CSS way for now, because setTimeout I am sometimes wary of (it's only 500ms though so it's not a huge deal). We can swap it over if it makes no difference.

Reply:
I don't like setTimeout either and I didn't know the CSS way, thanks for addressing!